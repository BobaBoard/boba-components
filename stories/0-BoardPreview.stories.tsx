import React from "react";
//import { linkTo } from "@storybook/addon-links";
import BoardPreview, { DisplayStyle } from "../src/board/BoardPreview";
import BoardsDisplay from "../src/board/BoardsDisplay";
import Tag from "../src/common/Tag";

import goreBackground from "./images/gore.png";
import anime from "./images/anime.png";
import crack from "./images/crack.png";
import meta from "./images/meta.png";
import book from "./images/book.png";
import villains from "./images/villains.png";
export default {
  title: "Board Preview",
  component: BoardPreview,
};

export const BoardPreviewSimple = () => (
  <BoardPreview
    slug="gore"
    avatar={`/${goreBackground}`}
    description="Love me some bruised bois (and more)."
    onClick={() => console.log("go!")}
  ></BoardPreview>
);

BoardPreviewSimple.story = {
  name: "simple",
};

export const BoardPreviewCompact = () => (
  <div style={{ width: "500px" }}>
    <div>
      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
        displayStyle={DisplayStyle.COMPACT}
      ></BoardPreview>
    </div>
    <div>
      <BoardPreview
        slug="thisisalongslug"
        avatar={`/${goreBackground}`}
        description="haveyoueverseen a description this long? oh shit boy here we come. Hey whaddup!"
        onClick={() => console.log("go!")}
        displayStyle={DisplayStyle.COMPACT}
      ></BoardPreview>
    </div>
  </div>
);

BoardPreviewCompact.story = {
  name: "compact",
};

export const BoardPreviewMini = () => (
  <BoardPreview
    slug="gore"
    avatar={`/${goreBackground}`}
    description="Love me some bruised bois (and more)."
    onClick={() => console.log("go!")}
    displayStyle={DisplayStyle.MINI}
  ></BoardPreview>
);

BoardPreviewMini.story = {
  name: "mini",
};

export const BoardPreviewWithUpdates = () => (
  <div style={{ width: "500px" }}>
    <BoardPreview
      slug="gore"
      avatar={`/${goreBackground}`}
      description="Love me some bruised bois (and more)."
      onClick={() => console.log("go!")}
      updates={5}
      backgroundColor="#333"
      color="#ff0f4b"
      href="#thisIsATest"
    />
    <BoardPreview
      slug="gore"
      avatar={`/${goreBackground}`}
      description="Love me some bruised bois (and more)."
      onClick={() => console.log("go!")}
      displayStyle={DisplayStyle.COMPACT}
      updates={5}
      color="#ff0f4b"
      href="#thisIsATest"
    />
    <BoardPreview
      slug="gore"
      avatar={`/${goreBackground}`}
      description="Love me some bruised bois (and more)."
      onClick={() => console.log("go!")}
      displayStyle={DisplayStyle.MINI}
      updates={5}
      color="#ff0f4b"
      href="#thisIsATest"
    />
  </div>
);

BoardPreviewWithUpdates.story = {
  name: "with updates",
};

export const BoardPreviewWithTags = () => (
  <div
    style={{
      display: "flex",
      alignItems: "end",
      justifyContent: "space-evenly",
    }}
  >
    <BoardPreview
      slug="gore"
      avatar={`/${goreBackground}`}
      description="Love me some bruised bois (and more)."
      onClick={() => console.log("go!")}
    >
      <Tag name="blood" color="#f96680" />
      <Tag name="knifeplay" color="#93b3b0" />
      <Tag name="aesthetic" color="#24d282" />
    </BoardPreview>

    <BoardPreview
      slug="gore"
      avatar={`/${goreBackground}`}
      description="Love me some bruised bois (and more)."
      onClick={() => console.log("go!")}
      displayStyle={DisplayStyle.COMPACT}
    >
      <Tag name="blood" color="#f96680" />
      <Tag name="knifeplay" color="#93b3b0" />
      <Tag name="aesthetic" color="#24d282" />
    </BoardPreview>
  </div>
);

BoardPreviewWithTags.story = {
  name: "with tags",
};

export const BoardPreviewWithColors = () => (
  <div>
    <div>
      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
      >
        <Tag name="blood" color="#f96680" />
        <Tag name="knifeplay" color="#93b3b0" />
        <Tag name="aesthetic" color="#24d282" />
      </BoardPreview>

      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
        displayStyle={DisplayStyle.COMPACT}
      >
        <Tag name="blood" color="#f96680" />
        <Tag name="knifeplay" color="#93b3b0" />
        <Tag name="aesthetic" color="#24d282" />
      </BoardPreview>
    </div>
    <div>
      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
        color="#f96680"
      >
        <Tag name="blood" color="#f96680" />
        <Tag name="knifeplay" color="#93b3b0" />
        <Tag name="aesthetic" color="#24d282" />
      </BoardPreview>

      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
        displayStyle={DisplayStyle.COMPACT}
        color="#f96680"
      >
        <Tag name="blood" color="#f96680" />
        <Tag name="knifeplay" color="#93b3b0" />
        <Tag name="aesthetic" color="#24d282" />
      </BoardPreview>
    </div>
    <div>
      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
        color="#24d282"
      >
        <Tag name="blood" color="#f96680" />
        <Tag name="knifeplay" color="#93b3b0" />
        <Tag name="aesthetic" color="#24d282" />
      </BoardPreview>

      <BoardPreview
        slug="gore"
        avatar={`/${goreBackground}`}
        description="Love me some bruised bois (and more)."
        onClick={() => console.log("go!")}
        displayStyle={DisplayStyle.COMPACT}
        color="#24d282"
      >
        <Tag name="blood" color="#f96680" />
        <Tag name="knifeplay" color="#93b3b0" />
        <Tag name="aesthetic" color="#24d282" />
      </BoardPreview>
    </div>
    <style jsx>{`
      div > div {
        display: flex;
        align-items: end;
        justify-content: space-evenly;
        margin: 50px 0;
      }
    `}</style>
  </div>
);

BoardPreviewWithColors.story = {
  name: "with colors",
};

export const BoardsDisplayStory = () => (
  <div style={{ width: "500px" }}>
    <BoardsDisplay
      title="Boards"
      boards={[
        {
          slug: "gore",
          avatar: "/" + goreBackground,
          description: "Love me some bruised bois (and more).",
          color: "#f96680",
        },
        {
          slug: "anime",
          avatar: "/" + anime,
          description: "We put the weeb in dweeb.",
          color: "#24d282",
          updates: 2,
        },
        {
          slug: "crack",
          avatar: "/" + crack,
          description: "What's crackalackin",
          color: "#f9e066",
          updates: 3,
        },
        {
          slug: "fic-club",
          avatar: "/" + book,
          description: "Come enjoy all the fics!",
          color: "#7724d2",
          updates: 5,
        },
        {
          slug: "meta",
          avatar: "/" + meta,
          description: "In My TiMeS wE CaLlEd It WaNk",
          color: "#f9e066",
        },
        {
          slug: "villain-thirst",
          avatar: "/" + villains,
          description: "Love to love 'em.",
          color: "#e22b4b",
        },
      ]}
      boardsDisplayStyle={DisplayStyle.MINI}
      onBoardClick={(slug) => console.log(slug)}
      getBoardHref={(slug) => `#${slug}`}
    />
  </div>
);
BoardsDisplayStory.story = {
  name: "boards display",
};
