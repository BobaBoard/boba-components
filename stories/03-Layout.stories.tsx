import React from "react";
//import { linkTo } from "@storybook/addon-links";
import Layout from "../src/Layout";
import { faBars } from "@fortawesome/free-solid-svg-icons";

export default {
  title: "Layout Preview",
  component: Layout,
};

export const SimpleButton = () => {
  return <Layout />;
};

SimpleButton.story = {
  name: "simple",
};
