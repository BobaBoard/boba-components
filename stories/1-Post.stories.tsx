import React from "react";
//import { linkTo } from "@storybook/addon-links";
import Card from "../src/common/Card";
import Header, { HeaderStyle } from "../src/post/Header";
import Footer from "../src/post/Footer";
import EditorFooter from "../src/post/EditorFooter";
import Post from "../src/post/Post";
import PostQuote from "../src/post/PostQuote";
// @ts-ignore
import Editor from "@bobaboard/boba-editor";

import oncelerAvatar from "./images/oncie.jpg";
import tuxedoAvatar from "./images/tuxedo-mask.jpg";
import mamoruAvatar from "./images/mamoru.png";

import oncieReaction from "./images/oncie-reaction.png";
import sportacusReaction from "./images/sportacus-reaction.png";
import luigiReaction from "./images/luigi-reaction.png";
import junkoReaction from "./images/junko-reaction.png";
import Button from "../src/common/Button";
import { action } from "@storybook/addon-actions";

export default {
  title: "Post Preview",
  component: Card,
};

export const CardSimple = () => (
  <div className="test-centering">
    <div className="container">
      <Card>
        <Editor
          editable={false}
          initialText={JSON.parse(
            '[{"insert":"Open RP"},{"attributes":{"header":1},"insert":"\\n"},{"insert":{"block-image":"https://cdn.discordapp.com/attachments/443967088118333442/691486081895628830/unknown.png"}}, {"attributes":{"italic":true},"insert":"You have my sword..."}]'
          )}
          onTextChange={() => {
            console.log("changed!");
          }}
          focus={true}
          onIsEmptyChange={() => {
            console.log("empty!");
          }}
          onSubmit={() => {
            // This is for cmd + enter pressed while in the editor
            console.log("submit!");
          }}
        />
      </Card>
    </div>
    <div className="container large">
      <Card>
        <Editor
          editable={false}
          initialText={JSON.parse(
            '[{"insert":"Open RP"},{"attributes":{"header":1},"insert":"\\n"},{"insert":{"block-image":"https://cdn.discordapp.com/attachments/443967088118333442/691486081895628830/unknown.png"}}, {"attributes":{"italic":true},"insert":"You have my sword..."}]'
          )}
          onTextChange={() => {
            console.log("changed!");
          }}
          focus={true}
          onIsEmptyChange={() => {
            console.log("empty!");
          }}
          onSubmit={() => {
            // This is for cmd + enter pressed while in the editor
            console.log("submit!");
          }}
        />
      </Card>
    </div>
    <div className="container">
      <Card>
        <Editor
          editable={false}
          initialText={JSON.parse(
            '[{"insert":"This card has little content."}]'
          )}
          onTextChange={() => {
            console.log("changed!");
          }}
          focus={true}
          onIsEmptyChange={() => {
            console.log("empty!");
          }}
          onSubmit={() => {
            // This is for cmd + enter pressed while in the editor
            console.log("submit!");
          }}
        />
      </Card>
    </div>
    <div className="container">
      <Card>
        <Editor
          editable={false}
          initialText={JSON.parse(
            '[{"insert":"This card has a really long word: JugemuJugemuGokonoSurikireKaijarisuigyonoSuigyomatsuUnraimatsuFuraimatsuKuNeruTokoroniSumuTokoroYaburaKojinoBuraKojiPaipopaipoPaiponoShuringanShuringannoGurindaiGurindainoPonpokopinoPonpokonanoChokyumeinoChosuke."}]'
          )}
          onTextChange={() => {
            console.log("changed!");
          }}
          focus={true}
          onIsEmptyChange={() => {
            console.log("empty!");
          }}
          onSubmit={() => {
            // This is for cmd + enter pressed while in the editor
            console.log("submit!");
          }}
        />
      </Card>
    </div>
    <div className="container">
      <Card height={300}>
        <Editor
          editable={false}
          initialText={JSON.parse(
            '[{"insert":"Open RP"},{"attributes":{"header":1},"insert":"\\n"},{"insert":{"block-image":"https://cdn.discordapp.com/attachments/443967088118333442/691486081895628830/unknown.png"}}, {"attributes":{"italic":true},"insert":"You have my sword..."}]'
          )}
          onTextChange={() => {
            console.log("changed!");
          }}
          focus={true}
          onIsEmptyChange={() => {
            console.log("empty!");
          }}
          onSubmit={() => {
            // This is for cmd + enter pressed while in the editor
            console.log("submit!");
          }}
        />
      </Card>
    </div>
    <style jsx>
      {`
        .test-centering {
          display: flex;
          flex-direction: column;
          align-items: center;
        }
        .container {
          max-width: 450px;
          margin-bottom: 10px;
          width: 100%;
        }
        .container.large {
          max-width: 800px;
        }
      `}
    </style>
  </div>
);

CardSimple.story = {
  name: "content card",
};

const FooterTemplate = (args: any) => (
  <>
    <div style={{ maxWidth: "500px" }}>
      <Footer {...args} />
    </div>
    <div style={{ maxWidth: "250px" }}>
      <Footer {...args} />
    </div>
  </>
);

// To show this story by itself, add export in front of const
const AnswerableFooter = FooterTemplate.bind({});
AnswerableFooter.args = {
  answerable: true,
};

const FooterWithNotes = FooterTemplate.bind({});
FooterWithNotes.args = {
  answerable: true,
  totalContributions: 5,
  directContributions: 2,
  totalComments: 4,
};

const FooterWithUpdates = FooterTemplate.bind({});
FooterWithUpdates.args = {
  ...FooterWithNotes.args,
  newContributions: 1,
  newComments: 3,
};

const FooterWithHighCount = FooterTemplate.bind({});
FooterWithHighCount.args = {
  answerable: true,
  totalContributions: 305,
  directContributions: 200,
  totalComments: 690,
  newContributions: 122,
  newComments: 300,
};

const NonAnswerableFooter = FooterTemplate.bind({});
NonAnswerableFooter.args = {
  ...FooterWithUpdates.args,
  answerable: false,
};

const FooterWithoutHref = FooterTemplate.bind({});
FooterWithoutHref.args = {
  ...FooterWithUpdates.args,
  notesLink: {
    onClick: action("withoutHref"),
  },
};

const FooterWithHref = FooterTemplate.bind({});
FooterWithHref.args = {
  ...FooterWithUpdates.args,
  answerable: false,
  notesLink: {
    onClick: action("withHref"),
    href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
};

export const FooterStory = (args: any) => (
  <div>
    <h2>Empty</h2>
    <AnswerableFooter {...AnswerableFooter.args} />
    <h2>No Updates</h2>
    <FooterWithNotes {...FooterWithNotes.args} />
    <h2>With Updates</h2>
    <FooterWithUpdates {...FooterWithUpdates.args} />
    <h2>With High Counts</h2>
    <FooterWithHighCount {...FooterWithHighCount.args} />
    <h2>Non Answerable</h2>
    <NonAnswerableFooter {...NonAnswerableFooter.args} />
    <h2>Click With Href</h2>
    <FooterWithHref {...FooterWithHref.args} />
    <h2>Click Without Href</h2>
    <FooterWithoutHref {...FooterWithoutHref.args} />
    <style jsx>
      {`
        h2 {
          color: white;
        }
        div {
          margin-top: 25px;
        }
        div > :global(div) {
          margin-top: 15px;
          background-color: pink;
        }
      `}
    </style>
  </div>
);

FooterStory.story = {
  name: "footer",
};

export const EditorFooterStory = () => (
  <div>
    <h2>Editor Footer</h2>
    <div style={{ maxWidth: "250px", backgroundColor: "turquoise" }}>
      <EditorFooter />
      <EditorFooter submittable={false} />
      <EditorFooter cancellable={false} />
    </div>
  </div>
);
EditorFooterStory.story = {
  name: "editor footer",
};

const HeaderTemplate = (args: any) => <Header {...args} />;

// To show this story by itself, add export in front of const
const RegularHeader = HeaderTemplate.bind({});
RegularHeader.args = {
  secretIdentity: {
    name: "Good Onceler",
    avatar: `/${oncelerAvatar}`,
  },
  createdMessage: "posted on: 2019/06/19 at 4:20pm",
  createdMessageLink: {
    onClick: action("withHref"),
    href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  size: HeaderStyle.REGULAR,
};

const CompactHeader = HeaderTemplate.bind({});
CompactHeader.args = {
  ...RegularHeader.args,
  size: HeaderStyle.COMPACT,
};

const WithUserIdentity = HeaderTemplate.bind({});
WithUserIdentity.args = {
  ...RegularHeader.args,
  secretIdentity: { name: "Tuxedo Mask", avatar: `/${tuxedoAvatar}` },
  userIdentity: { name: "SexyDaddy69", avatar: `/${mamoruAvatar}` },
};

const WithUserIdentityCompact = HeaderTemplate.bind({});
WithUserIdentityCompact.args = {
  ...WithUserIdentity.args,
  size: HeaderStyle.COMPACT,
};

const WithUserIdentityForceHide = HeaderTemplate.bind({});
WithUserIdentityForceHide.args = {
  ...WithUserIdentity.args,
  forceHide: true,
};
const WithUserIdentityForceHideCompact = HeaderTemplate.bind({});
WithUserIdentityForceHideCompact.args = {
  ...WithUserIdentityCompact.args,
  forceHide: true,
};

export const HeaderStory = () => (
  <div>
    <RegularHeader {...RegularHeader.args} />
    <CompactHeader {...CompactHeader.args} />
    <WithUserIdentity {...WithUserIdentity.args} />
    <WithUserIdentityCompact {...WithUserIdentityCompact.args} />
    <WithUserIdentityForceHide {...WithUserIdentityForceHide.args} />
    <WithUserIdentityForceHideCompact
      {...WithUserIdentityForceHideCompact.args}
    />
    {/*These will probably need to be deleted at some point*/}
    <RegularHeader {...RegularHeader.args} newPost />
    <RegularHeader {...RegularHeader.args} newComments />
    <RegularHeader {...RegularHeader.args} newComments newContributions />

    <div style={{ width: "200px", backgroundColor: "green" }}>
      <WithUserIdentity {...WithUserIdentity.args} />
    </div>
    <div style={{ width: "200px", backgroundColor: "yellow" }}>
      <WithUserIdentity
        {...WithUserIdentity.args}
        newComments
        newContributions
      />
    </div>
    <style jsx>
      {`
        div {
          max-width: 500px;
          margin-top: 25px;
          background-color: white;
        }
        div > :global(div) {
          margin-top: 15px;
        }
      `}
    </style>
  </div>
);

HeaderStory.story = {
  name: "header",
};

const PostTemplate = (args: any) => <Post {...args} />;
export const NonEditable = PostTemplate.bind({});
NonEditable.story = {
  name: "non-editable post",
};

NonEditable.args = {
  createdTime: "2019/05/14 at 7:34pm",
  text:
    '[{"insert":"Open RP"},{"attributes":{"header":1},"insert":"\\n"},{"insert":{"block-image":"https://cdn.discordapp.com/attachments/443967088118333442/691486081895628830/unknown.png"}}, {"attributes":{"italic":true},"insert":"You have my sword..."}]',
  secretIdentity: { name: "Tuxedo Mask", avatar: `/${tuxedoAvatar}` },
  userIdentity: { name: "SexyDaddy69", avatar: `/${mamoruAvatar}` },
  onNewContribution: () => action("newContribution"),
  onNewComment: () => action("newComment"),
  onNotesClick: () => console.log("click"),
  createdTimeLink: {
    onClick: action("createdTime"),
    href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  notesLink: {
    onClick: action("notesLink"),
    href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
};

export const UpdatedPost = PostTemplate.bind({});
UpdatedPost.story = {
  name: "updated post",
};
UpdatedPost.args = {
  ...NonEditable.args,
  totalContributions: 15,
  directContributions: 3,
  totalComments: 5,
  newComments: 3,
  newContributions: 5,
};

export const AnswerablePost = PostTemplate.bind({});
AnswerablePost.story = {
  name: "answerable post",
};
AnswerablePost.args = {
  ...UpdatedPost.args,
  answerable: true,
};

export const TaggedPost = PostTemplate.bind({});
TaggedPost.story = {
  name: "tagged post",
};
TaggedPost.args = {
  ...AnswerablePost.args,
  tags: {
    indexTags: ["indexable"],
    categoryTags: ["category"],
    contentWarnings: ["bad content"],
    whisperTags: [
      "tag1",
      "tag2",
      "a long tag",
      "a very very very very very long tag with many words",
      "JugemuJugemuGokonoSurikireKaijarisuigyonoSuigyomatsuUnraimatsuFuraimatsuKuNeruTokoroniSumuTokoroYaburaKojinoBuraKojiPaipopaipoPaiponoShuringanShuringannoGurindaiGurindainoPonpokopinoPonpokonanoChokyumeinoChosuke",
    ],
  },
  answerable: true,
};

export const ReactablePost = PostTemplate.bind({});
ReactablePost.story = {
  name: "reactable post",
};
ReactablePost.args = {
  ...TaggedPost.args,
  reactable: true,
  reactions: [
    { image: oncieReaction, count: 3 },
    { image: sportacusReaction, count: 6 },
    { image: luigiReaction, count: 11 },
    { image: junkoReaction, count: 20 },
  ],
};

export const ActionPost = PostTemplate.bind({});
ActionPost.story = {
  name: "actionable post",
};
ActionPost.args = {
  ...TaggedPost.args,
  menuOptions: [
    {
      name: "Copy Link",
      link: {
        onClick: action("copy!"),
        href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
      },
    },
  ],
};

export const HighlightPost = () => {
  const postRef = React.createRef<any>();
  return (
    <div>
      <Post {...TaggedPost.args} ref={postRef} />
      <div style={{ marginTop: "20px" }}>
        <Button
          onClick={() => {
            action("highlight")(postRef.current);
            postRef.current.highlight("red");
          }}
        >
          Highlight!
        </Button>
      </div>
    </div>
  );
};

HighlightPost.story = {
  name: "highlight post",
};

export const PostQuoteStory = () => {
  return (
    <div style={{ width: "300px" }}>
      <PostQuote
        createdTime="yesterday"
        text={
          '[{"insert":"Open RP"},{"attributes":{"header":1},"insert":"\\n"},{"insert":{"block-image":"https://cdn.discordapp.com/attachments/443967088118333442/691486081895628830/unknown.png"}}, {"attributes":{"italic":true},"insert":"You have my sword..."}]'
        }
        secretIdentity={{ name: "Tuxedo Mask", avatar: `/${tuxedoAvatar}` }}
        userIdentity={{ name: "SexyDaddy69", avatar: `/${mamoruAvatar}` }}
      />
    </div>
  );
};
PostQuoteStory.story = {
  name: "quote",
};
