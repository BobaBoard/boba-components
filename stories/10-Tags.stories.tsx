import React from "react";
import Tags from "../src/post/Tags";
import CategoryFilter from "../src/common/CategoryFilter";

export default {
  title: "Tags Preview",
  component: Tags,
};

export const EditableTags = () => {
  const [tags, setTags] = React.useState([
    { name: "this is an indexable tag", color: "#19a4e6", indexable: true },
    { name: "another one", color: "#ce769c", indexable: true },
    { name: "tag1" },
    { name: "tag2" },
    { name: "tag1234561789" },
    { name: "tag1234561789x2" },
  ]);
  return (
    <div style={{ width: "500px", backgroundColor: "white" }}>
      <Tags
        tags={tags}
        onTagsAdd={(tag) => setTags([...tags, tag])}
        onTagsDelete={(tag) => {
          setTags(tags.filter((currentTag) => currentTag != tag));
        }}
        editable
        onSubmit={() => console.log("submit!!")}
      />
    </div>
  );
};

EditableTags.story = {
  name: "editable",
};

export const DisplayTags = () => {
  const [tags, setTags] = React.useState([
    { name: "this is an indexable tag", color: "#19a4e6", indexable: true },
    { name: "another one", color: "#ce769c", indexable: true },
    { name: "tag1" },
    { name: "tag2" },
    { name: "a long tag" },
    { name: "a very very very very very long tag with many words" },
    {
      name:
        "a tag with many words that is actually more than one single line long used to check words splitting",
    },
    {
      name:
        "JugemuJugemuGokonoSurikireKaijarisuigyonoSuigyomatsuUnraimatsuFuraimatsuKuNeruTokoroniSumuTokoroYaburaKojinoBuraKojiPaipopaipoPaiponoShuringanShuringannoGurindaiGurindainoPonpokopinoPonpokonanoChokyumeinoChosuke",
    },
  ]);
  return (
    <div style={{ width: "500px", backgroundColor: "white" }}>
      <Tags tags={tags} />
    </div>
  );
};

DisplayTags.story = {
  name: "non editable",
};

export const FilterTags = () => {
  const [tags, setTags] = React.useState([
    { name: "tag1", active: true },
    { name: "tag2", active: false },
    { name: "a long tag", active: false },
    {
      name: "a very very very very very long tag with many words",
      active: false,
    },
    {
      name:
        "a tag with many words that is actually more than one single line long used to check words splitting",
      active: true,
    },
    {
      name:
        "JugemuJugemuGokonoSurikireKaijarisuigyonoSuigyomatsuUnraimatsuFuraimatsuKuNeruTokoroniSumuTokoroYaburaKojinoBuraKojiPaipopaipoPaiponoShuringanShuringannoGurindaiGurindainoPonpokopinoPonpokonanoChokyumeinoChosuke",
      active: false,
    },
  ]);
  return (
    <div style={{ width: "500px", backgroundColor: "white" }}>
      <CategoryFilter
        categories={tags}
        onCategoryStateChange={(name, active) => {
          setTags(
            tags.map((tag) => (tag.name == name ? { name, active } : tag))
          );
        }}
      />
    </div>
  );
};

FilterTags.story = {
  name: "category filter",
};
