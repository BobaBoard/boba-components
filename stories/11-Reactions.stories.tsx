import React from "react";
import Reaction from "../src/common/Reaction";

import oncieReaction from "./images/oncie-reaction.png";

export default {
  title: "Reactions Preview",
  component: Reaction,
};

export const ReactionStory = () => {
  const [count, setCount] = React.useState(1);
  return (
    <div style={{ width: "500px", backgroundColor: "white", padding: "15px" }}>
      <Reaction
        image={oncieReaction}
        count={count}
        onClick={() => setCount(count + 1)}
      />
    </div>
  );
};

ReactionStory.story = {
  name: "reaction",
};
