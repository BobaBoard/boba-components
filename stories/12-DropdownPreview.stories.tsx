import React from "react";
import DropdownListMenu from "../src/common/DropdownListMenu";
import Button from "../src/buttons/Button";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCaretDown,
  faMapPin,
  faVolumeMute,
  faBellSlash,
  faPaintBrush,
} from "@fortawesome/free-solid-svg-icons";
import { action } from "@storybook/addon-actions";

export default {
  title: "Dropdown Preview",
  component: DropdownListMenu,
};

export const ClassicDropdownStory = () => {
  return (
    <div style={{ width: "500px", backgroundColor: "white", padding: "15px" }}>
      <DropdownListMenu
        options={[
          {
            name: "Pin board",
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Mute board",
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Dismiss notifications",
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Customize Summary",
            link: {
              onClick: action("noHrefClick"),
            },
          },
        ]}
        zIndex={51}
      >
        <Button>Click me!</Button>
      </DropdownListMenu>
    </div>
  );
};

ClassicDropdownStory.story = {
  name: "classic",
};

export const IconDropwdownStory = () => {
  return (
    <div style={{ width: "500px", backgroundColor: "white", padding: "15px" }}>
      <DropdownListMenu
        options={[
          {
            name: "Pin board",
            icon: faMapPin,
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Mute board",
            icon: faVolumeMute,
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Dismiss notifications",
            icon: faBellSlash,
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Customize Summary",
            icon: faPaintBrush,
            link: {
              onClick: action("noHrefClick"),
            },
          },
        ]}
        zIndex={51}
      >
        <FontAwesomeIcon icon={faCaretDown} />
      </DropdownListMenu>
    </div>
  );
};

IconDropwdownStory.story = {
  name: "icon",
};

export const MultipleDropdowns = () => {
  return (
    <div style={{ width: "500px", backgroundColor: "white", padding: "15px" }}>
      <DropdownListMenu
        options={[
          {
            name: "Pin board",
            icon: faMapPin,
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Inner Options",
            icon: faVolumeMute,
            options: [
              {
                name: "Inner Option 1",
                icon: faBellSlash,
                link: {
                  onClick: action("noHrefClick"),
                },
              },
              {
                name: "Inner Option 2",
                icon: faPaintBrush,
                options: [
                  {
                    name: "Inner Option 21",
                    icon: faBellSlash,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                  {
                    name: "Inner Option 22",
                    icon: faPaintBrush,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                  {
                    name: "Inner Option 23",
                    icon: faPaintBrush,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                  {
                    name: "Inner Option 2424242424242424242424",
                    icon: faPaintBrush,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                ],
              },
            ],
          },
          {
            name: "Dismiss notifications",
            icon: faBellSlash,
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Customize Summary",
            icon: faPaintBrush,
            link: {
              onClick: action("noHrefClick"),
            },
          },
        ]}
        zIndex={51}
      >
        <FontAwesomeIcon icon={faCaretDown} />
      </DropdownListMenu>
    </div>
  );
};

MultipleDropdowns.story = {
  name: "multiple",
};

export const WithHeader = () => {
  return (
    <div style={{ width: "500px", backgroundColor: "white", padding: "15px" }}>
      <DropdownListMenu
        options={[
          {
            name: "OnClick Only",
            icon: faMapPin,
            link: {
              onClick: action("noHrefClick"),
            },
          },
          {
            name: "Inner Options",
            icon: faVolumeMute,
            options: [
              {
                name: "Inner Option 1",
                icon: faBellSlash,
                link: {
                  onClick: action("noHrefClick"),
                },
              },
              {
                name: "Inner Option 2",
                icon: faPaintBrush,
                options: [
                  {
                    name: "Inner Option 21",
                    icon: faBellSlash,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                  {
                    name: "Inner Option 22",
                    icon: faPaintBrush,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                  {
                    name: "Inner Option 23",
                    icon: faPaintBrush,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                  {
                    name: "Inner Option 2424242424242424242424",
                    icon: faPaintBrush,
                    link: {
                      onClick: action("noHrefClick"),
                    },
                  },
                ],
              },
            ],
          },
          {
            name: "Href Only",
            icon: faBellSlash,
            link: {
              href: "#href",
            },
          },
          {
            name: "Both",
            icon: faPaintBrush,
            link: {
              href: "#href",
              onClick: action("noHrefClick"),
            },
          },
        ]}
        zIndex={51}
      >
        <DropdownListMenu.Header>
          <div style={{ backgroundColor: "red" }}>
            This is a header, this is a header, a beautiful header
          </div>
        </DropdownListMenu.Header>
        <FontAwesomeIcon icon={faCaretDown} />
      </DropdownListMenu>
    </div>
  );
};

WithHeader.story = {
  name: "with header",
};
