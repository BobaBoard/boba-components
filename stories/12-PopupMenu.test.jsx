import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme'; 

import { ReactionStory } from './12-PopupMenu.stories';

configure({adapter: new Adapter()});

describe('ReactionStory', () => {
  const wrapper = mount(<ReactionStory/>);

  it('should render with button', () => {
    expect(wrapper.find('button')).toHaveLength(1);
  });

})
