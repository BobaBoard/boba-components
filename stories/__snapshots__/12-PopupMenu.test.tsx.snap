// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionStory should render 1`] = `ShallowWrapper {}`;
